src/libPMacc/include/cuSTL/container/CartBuffer.hpp

```diff
@@ -56,12 +62,13 @@ namespace container
  */
 template<typename Type, int T_dim, typename Allocator = allocator::EmptyAllocator,
                                   typename Copier = mpl::void_,
-                                  typename Assigner = mpl::void_>
-class CartBuffer
+                                  typename Assigner = bmpl::vector<bmpl::_1, bmpl::_2> >
+class CartBuffer : public
+    bmpl::apply<Assigner, bmpl::int_<T_dim>, CartBuffer<Type, T_dim, Allocator, Copier, Assigner> >::type
 {
 public:
     typedef Type type;
-    typedef CartBuffer<Type, T_dim, Allocator, Copier, Assigner> This;
+    typedef CartBuffer This;
```

Comment:
`This` usages could be completely removed as this typedef just obscures the usages...